src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -1123,7 +1125,7 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
             }
           } else None
         case (overridden, env) =>
-          val om = specializedOverload(clazz, overridden, env)
+          val om = specializedOverload(clazz, overridden, env, excludeFlags = overriding.isPublic)
```

Comment:
Would this work instead, without changing the method `specializedOverload`?
```suggestion
          val om = specializedOverload(clazz, overriding, env)
```

Reply:
Yes, I've tried, but failed, because there is still exists `protected local`